app/controllers/search_controller.rb

```diff
@@ -5,7 +5,11 @@ def index
     if @form.valid?
       @result = FirmRepository.new.search(@form.to_query, page: page)
     else
-      render 'landing_page/show'
+      if params[:origin] == 'results'
```

Comment:
Move this to a private predicate like `from_results?` and then add another private method the does the check against that predicate and returns the correct template, so here you could say just `render the_method_you_just_made` instead of branching in here.